src/scripts/JIRA/etl-437/main.py

```diff
@@ -0,0 +1,136 @@
+"""
+This script copies all objects under an S3 prefix which begin with
+a specific string to an object in the same "directory" but with
+the basename prefaced with another string.
+"""
+import json
+import os
+import argparse
+import boto3
+
+def read_args():
+    parser = argparse.ArgumentParser(
+            description="Copy S3 objects to another object prefaced with a string.")
+    parser.add_argument(
+            "--s3-bucket",
+            required=True,
+            help="The name of the S3 bucket."
+    )
+    parser.add_argument(
+            "--s3-prefix",
+            required=True,
+            help="The S3 prefix under which qualifying objects will be copied."
+    )
+    parser.add_argument(
+            "--match-string",
+            required=True,
+            help="Objects will be copied if they begin with this string."
+    )
+    parser.add_argument(
+            "--preface-string",
+            required=True,
+            help="Objects which are copied will be prefaced with this string."
+    )
+    parser.add_argument(
+            "--profile",
+            help="Optional. The local AWS profile to use."
+    )
+    args = parser.parse_args()
+    return args
+
+def get_matching_objects(
+        s3_client: "botocore.client.S3",
+        bucket: str,
+        key_prefix: str,
+        match_string: str) -> "list[str]":
```

Comment:
Nit: I think for later versions of Python you can probably just do

```
-> list[str]
# but for older versions
from typing import List
-> List[str]
```

This is the same for dictionaries, but not a huge concern.